llvm/test/tools/llvm-symbolizer/skip-line-zero.s

```diff
@@ -0,0 +1,252 @@
+	.byte	11                              # DW_AT_byte_size
+	.byte	11                              # DW_FORM_data1
+	.byte	0                               # EOM(1)
+	.byte	0                               # EOM(2)
+	.byte	0                               # EOM(3)
+	.section	.debug_info,"",@progbits
+.Lcu_begin0:
+	.long	.Ldebug_info_end0-.Ldebug_info_start0 # Length of Unit
+.Ldebug_info_start0:
+	.short	4                               # DWARF version number
+	.long	.debug_abbrev                   # Offset Into Abbrev. Section
+	.byte	8                               # Address Size (in bytes)
+	.byte	1                               # Abbrev [1] 0xb:0x55 DW_TAG_compile_unit
+	.long	.Linfo_string0                  # DW_AT_producer
+	.short	29                              # DW_AT_language
+	.long	.Linfo_string1                  # DW_AT_name
+	.long	.Lline_table_start0             # DW_AT_stmt_list
+	.quad	0                               # DW_AT_low_pc
+	.long	.Ldebug_ranges0                 # DW_AT_ranges
+	.byte	2                               # Abbrev [2] 0x26:0x19 DW_TAG_subprogram
+	.quad	0x1710                          # DW_AT_low_pc (.Lfunc_begin0)
+	.long	0x171b-0x1710                   # DW_AT_high_pc(.Lfunc_end0-.Lfunc_begin0)
+	.byte	1                               # DW_AT_frame_base
+	.byte	86
+	.long	.Linfo_string2                  # DW_AT_name
+	.byte	1                               # DW_AT_decl_file
+	.byte	1                               # DW_AT_decl_line
+	.long	88                              # DW_AT_type
+                                        # DW_AT_external
+	.byte	3                               # Abbrev [3] 0x3f:0x19 DW_TAG_subprogram
+	.quad	0x16c0                          # DW_AT_low_pc (.Lfunc_begin1)
+	.long	0x16df-0x16c0                   # DW_AT_high_pc (.Lfunc_end1-.Lfunc_begin1)
+	.byte	1                               # DW_AT_frame_base
+	.byte	86
+	.long	.Linfo_string4                  # DW_AT_name
+	.byte	1                               # DW_AT_decl_file
+	.byte	2                               # DW_AT_decl_line
+                                        # DW_AT_prototyped
+	.long	88                              # DW_AT_type
+                                        # DW_AT_external
+	.byte	4                               # Abbrev [4] 0x58:0x7 DW_TAG_base_type
+	.long	.Linfo_string3                  # DW_AT_name
+	.byte	5                               # DW_AT_encoding
+	.byte	4                               # DW_AT_byte_size
+	.byte	0                               # End Of Children Mark
+.Ldebug_info_end0:
+	.section	.debug_ranges,"",@progbits
+.Ldebug_ranges0:
+	.quad	0x1710                          #.Lfunc_begin0
+	.quad	0x171b                          #.Lfunc_end0
+	.quad	0x16c0                          #.Lfunc_begin1
+	.quad	0x16df                          #.Lfunc_end1
+	.quad	0
+	.quad	0
+	.section	.debug_str,"MS",@progbits,1
+.Linfo_string0:
+	.asciz	""                              # string offset=0
```

Comment:
Having said this is fine I think this empty string makes it a bit harder to read because it looks interesting and I'm matching against what I expect the dwarf to look like. I suggest either:

1. Put in a short dummy string to indicate this is the producer, e.g. "Clang 3.21"

2. Edit the Dwarf a bit more to be valid but shorter:
2.1 Remove these lines from .debug_abbrev:
	.byte	37                              # DW_AT_producer
	.byte	14                              # DW_FORM_strp
	.byte	19                              # DW_AT_language
	.byte	5                               # DW_FORM_data2
	.byte	3                               # DW_AT_name
	.byte	14                              # DW_FORM_strp
2.2 Remove these lines form .debug_info:
	.long	.Linfo_string0                  # DW_AT_producer
	.short	29                              # DW_AT_language
	.long	.Linfo_string1                  # DW_AT_name
2.3 Remove these lines from .debug_str:
.Linfo_string0:
	.asciz	""                              # string offset=0
.Linfo_string1:
	.asciz	"main.c"                        # string offset=113

Reply:
Done. Opted 2nd option.